src/User/UserManager.php

```diff
@@ -198,16 +198,38 @@ public function addRecord($name, $provider_user_id, $token, $additional_data = N
     ];
 
     $user_info = $this->entityTypeManager->getStorage('social_post')->create($values);
-    $user_info->setToken($token);
```

Comment:
I know there is an `user_id` initialized before the if checking if user exists. Move `$user_id = $this->getCurrentUser();` it after the if block.